(dp1
S'output'
p2
S"<type 'exceptions.AttributeError'> 'list' object has no attribute 'get'"
p3
sS'layer'
p4
S'/home/kitewarrior4/ITWS/assg3/web2py/applications/recipository/controllers/default.py'
p5
sS'code'
p6
S"# -*- coding: utf-8 -*-\n# this file is released under public domain and you can use without limitations\n\n#########################################################################\n## This is a sample controller\n## - index is the default action of any application\n## - user is required for authentication and authorization\n## - download is for downloading files uploaded in the db (does streaming)\n#########################################################################\n\ndef index():\n    records = SQLTABLE(db().select(db.recipe.ALL),\n                      upload=URL('download'),\n                      headers=['name','description','image'])\n    return locals()\n\n#@cache.action()\ndef download():\n    return response.download(request, db)\n\ndef user():\n    return dict(form=auth())\n\ndef call():\n    return service()\n\n\n@auth.requires_login()\ndef newrecipe():\n    recipe = SQLFORM(db.recipe,upload=URL('download')).process()\n    return locals()\nresponse._vars=response._caller(index)\n"
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "/home/kitewarrior4/ITWS/assg3/web2py/gluon/restricted.py", line 227, in restricted\n    exec ccode in environment\n  File "/home/kitewarrior4/ITWS/assg3/web2py/applications/recipository/controllers/default.py", line 32, in <module>\n  File "/home/kitewarrior4/ITWS/assg3/web2py/gluon/globals.py", line 412, in <lambda>\n    self._caller = lambda f: f()\n  File "/home/kitewarrior4/ITWS/assg3/web2py/applications/recipository/controllers/default.py", line 14, in index\n    headers=[\'name\',\'description\',\'image\'])\n  File "/home/kitewarrior4/ITWS/assg3/web2py/gluon/sqlhtml.py", line 3180, in __init__\n    if isinstance(headers.get(c, c), dict):\nAttributeError: \'list\' object has no attribute \'get\'\n'
p11
s.